cpp/src/arrow/builder.h

```diff
@@ -899,6 +899,31 @@ class ARROW_EXPORT DictionaryBuilder : public ArrayBuilder {
   int32_t byte_width_;
 };
 
+template <>
+class ARROW_EXPORT DictionaryBuilder<NullType> : public ArrayBuilder {
+ public:
+  ~DictionaryBuilder();
+
+  DictionaryBuilder(const std::shared_ptr<DataType>& type, MemoryPool* pool);
+  explicit DictionaryBuilder(MemoryPool* pool);
+
+  /// \brief Append a scalar value
+  // Status Append(const Scalar& value);
```

Comment:
Should this be commented out?

Reply:
Yes, this should be removed. You cannot add a scalar to a NullArray.